backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/nodes/ImportedPackageNodeDescriptionProvider.java

```diff
@@ -0,0 +1,135 @@
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.diagram.common.view.nodes;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+import org.eclipse.sirius.components.view.builder.IViewDiagramElementFinder;
+import org.eclipse.sirius.components.view.builder.providers.IColorProvider;
+import org.eclipse.sirius.components.view.diagram.DiagramDescription;
+import org.eclipse.sirius.components.view.diagram.HeaderSeparatorDisplayMode;
+import org.eclipse.sirius.components.view.diagram.InsideLabelDescription;
+import org.eclipse.sirius.components.view.diagram.InsideLabelPosition;
+import org.eclipse.sirius.components.view.diagram.InsideLabelStyle;
+import org.eclipse.sirius.components.view.diagram.LabelTextAlign;
+import org.eclipse.sirius.components.view.diagram.NodeDescription;
+import org.eclipse.sirius.components.view.diagram.NodePalette;
+import org.eclipse.sirius.components.view.diagram.NodeStyleDescription;
+import org.eclipse.sirius.components.view.diagram.NodeToolSection;
+import org.eclipse.sirius.components.view.diagram.SynchronizationPolicy;
+import org.eclipse.sirius.components.view.diagram.UserResizableDirection;
+import org.eclipse.syson.diagram.common.view.services.description.ToolDescriptionService;
+import org.eclipse.syson.sysml.SysmlPackage;
+import org.eclipse.syson.sysmlcustomnodes.SysMLCustomnodesFactory;
+import org.eclipse.syson.util.AQLUtils;
+import org.eclipse.syson.util.IDescriptionNameGenerator;
+import org.eclipse.syson.util.SysMLMetamodelHelper;
+import org.eclipse.syson.util.ViewConstants;
+
+/**
+ * Used to create the imported package node description in all diagrams.
+ *
+ * @author Jerome Gout
+ */
+public class ImportedPackageNodeDescriptionProvider extends AbstractNodeDescriptionProvider {
+
+    protected final IDescriptionNameGenerator nameGenerator;
+
+    private final ToolDescriptionService toolDescriptionService;
+
+    public ImportedPackageNodeDescriptionProvider(IColorProvider colorProvider, IDescriptionNameGenerator nameGenerator) {
+        super(colorProvider);
+        this.nameGenerator = Objects.requireNonNull(nameGenerator);
+        this.toolDescriptionService = new ToolDescriptionService(nameGenerator);
+    }
+
+    @Override
+    public NodeDescription create() {
+        String domainType = SysMLMetamodelHelper.buildQualifiedName(SysmlPackage.eINSTANCE.getNamespaceImport());
+        return this.diagramBuilderHelper.newNodeDescription()
+                .collapsible(true)
+                .defaultHeightExpression("100")
+                .defaultWidthExpression("100")
+                .domainType(domainType)
+                .insideLabel(this.createInsideLabelDescription())
+                .name(this.nameGenerator.getNodeName(SysmlPackage.eINSTANCE.getNamespaceImport()))
+                .semanticCandidatesExpression("aql:self.getAllReachable(" + domainType + ")")
```

Comment:
Please use AQLUtils here and everywhere else it's possible in this file

Reply:
Done